### ISSUE TITLE
Loading Phyloseq- R stops responding

### ISSUE THREAD

**Issue by locon833** [joey711/phyloseq#424]
I have recently updated my R software to the 3.1.2 version and am trying to install phyloseq to work with my Illumina data. I have the Mac software OS X 10.9.5. I am able to install the phyloseq package as well as its dependencies, but when I try to load phyloseq: library("phyloseq") my R software stops responding or it tells me that phyloseq is not available for R version 3.1.2. I'm using the the 3.0 release version of Bioconductor, which to my knowledge is the most recent release. If anyone knows how I can solve this issue so that I can work with phyloseq it would be greatly appreciated.

**Comment by joey711** [comment]
I saw a similar problem recently and I think it is related to Bioconductor installation in general.

What version of phyloseq are you using? Did you update your BioC installation or do a fresh install?

Also, are you using either the "Snow Leopard" or "Mavericks" version, by chance?

One option is to try and do a fresh installation of BioC.

I will update more here when/if I figure out what is going on. 

Note that there are no build/install issues reported on any of the BioC check servers:

http://bioconductor.org/checkResults/release/bioc-LATEST/phyloseq/

(which is frustrating to me because I did also encounter this issue transiently. Updating BioC fixed it for me)

**Comment by locon833** [comment]
Thanks for your response.

I'm using the phyloseq version 1.10.0. I did a fresh install of BioC using the steps for the quick install on the github page:

> source("http://bioconductor.org/biocLite.R")
Bioconductor version 3.0 (BiocInstaller 1.16.1), ?biocLite for help
> biocLite()
BioC_mirror: http://bioconductor.org
Using Bioconductor version 3.0 (BiocInstaller 1.16.1), R version 3.1.2.
Old packages: 'foreign', 'manipulate'
Update all/some/none? [a/s/n]: 
A

trying URL 'http://cran.rstudio.com/bin/macosx/mavericks/contrib/3.1/foreign_0.8-62.tgz'
Content type 'application/x-gzip' length 275657 bytes (269 Kb)
opened URL
==================================================
downloaded 269 Kb

trying URL 'http://cran.rstudio.com/bin/macosx/mavericks/contrib/3.1/manipulate_1.0.1.tgz'
Content type 'application/x-gzip' length 33362 bytes (32 Kb)
opened URL
==================================================
downloaded 32 Kb


The downloaded binary packages are in
	/var/folders/kt/_6vrl2sd4r7dyqz24y1mn02c0000gn/T//Rtmpq13kvp/downloaded_packages


I think was able to install phyloseq using:
> > biocLite("phyloseq")
BioC_mirror: http://bioconductor.org
Using Bioconductor version 3.0 (BiocInstaller 1.16.1), R version 3.1.2.
Installing package(s) 'phyloseq'
trying URL 'http://bioconductor.org/packages/3.0/bioc/bin/macosx/mavericks/contrib/3.1/phyloseq_1.10.0.tgz'
Content type 'application/x-gzip' length 3231971 bytes (3.1 Mb)
opened URL
==================================================
downloaded 3.1 Mb


I saw phyloseq in my available packages but then tried to load it and it crashed again. I'm fairly new to using R. I've only worked with QIIME and mothur so I'm not sure if I installed the BioC properly into R?

**Comment by joey711** [comment]
From what I can tell so far this is not a phyloseq bug, as phyloseq passes all checks on the official servers. It sounds like an R installation or Bioconductor installation bug. It might also be a "Mavericks" bug. There is a fork in the R binaries for Mac having to do with which C compiler is used, either snow-leopard or the mavericks one. I can't rule out yet this has something to do with it. You might consider installing the non-mavericks version of R for Mac, and installing your packages over again.

This is a very rare and strange problem. You might also consider posting it to the BioC Support Site to see if some people at BioC are already aware of this problem:

https://support.bioconductor.org/

Cheers

joey

p.s. I'll keep this open until we know have some kind of diagnosis...

**Comment by locon833** [comment]
Thanks, Joey.

I will download the snow leopard version of R and retry everything. I will let you know if that solves anything. Appreciate all of your help.

**Comment by locon833** [comment]
I re-downloaded the R package previous to both the snow leopard and mavericks version. I'm still running into the same problem. I will post to the BioC support page.

**Comment by joey711** [comment]
Great. Thanks. If you don't mind, would you link the support page post here once you have posted it? Also, please let me know when/if you have a solution. I'm guessing you're not the only one to have this problem. In several years of R development I have not seen this problem.

**Comment by locon833** [comment]
No problem, here is the link to the post: https://support.bioconductor.org/p/63998/

**Comment by locon833** [comment]
So I have an update, I was able to get phyloseq loaded without crashing the program by re-downloading the developer tools: XQuartz and XCode on my Mac. I'm now running into this issue when I try to complete the library("phyloseq") command: 

> library("phyloseq", lib.loc="/Library/Frameworks/R.framework/Versions/3.1/Resources/library")
Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object '/Library/Frameworks/R.framework/Versions/3.1/Resources/library/RcppArmadillo/libs/RcppArmadillo.so':
  dlopen(/Library/Frameworks/R.framework/Versions/3.1/Resources/library/RcppArmadillo/libs/RcppArmadillo.so, 6): Library not loaded: /Library/Frameworks/R.framework/Versions/3.1/Resources/lib/libgfortran.2.dylib
  Referenced from: /Library/Frameworks/R.framework/Versions/3.1/Resources/library/RcppArmadillo/libs/RcppArmadillo.so
  Reason: image not found
Error: package or namespace load failed for ‘phyloseq’

Any ideas how to fix this?

**Comment by thomasrime** [comment]
Hi all

I don't know if it helps (or if this thread is still actual...) but I had the same problem (couldn't install phyloseq with install.packages()). (on Mac OSX but don't know exactly the os... it's not my computer...)

I simply load it with biocLite("phyloseq"). This looks to trivial to be true but it works for me :) hope it helps!

good luck!

**Comment by spholmes** [comment]
Usually this problem occurs if you start a new session and forgot to add
the BioC repositories as well
as the cran ones;
Use setRepositories()
and choose all BioC repos as well as cran,


On Thu, Feb 26, 2015 at 4:21 AM, thomasrime <notifications@github.com>
wrote:

> Hi all
>
> I don't know if it helps (or if this thread is still actual...) but I had
> the same problem (couldn't install phyloseq with install.packages()). (on
> Mac OSX but don't know exactly the os... it's not my computer...)
>
> I simply load it with biocLite("phyloseq"). This looks to trivial to be
> true but it works for me :) hope it helps!
>
> good luck!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joey711/phyloseq/issues/424#issuecomment-76145889>.
>



-- 
Susan Holmes
Professor, Statistics and BioX
Director, MCS
Sequoia Hall,
390 Serra Mall
Stanford, CA 94305
http://www-stat.stanford.edu/~susan/

**Comment by joey711** [comment]
@thomasrime The official "release" version of phyloseq is through Bioconductor:

http://bioconductor.org/packages/devel/bioc/html/phyloseq.html

You can see there the recommended installation instructions for installing the latest "stable" release:

```S
source("http://bioconductor.org/biocLite.R")
biocLite("phyloseq")
```

These instructions are also listed elsewhere, including the phyloseq installation tutorial:

http://joey711.github.io/phyloseq/install

As for the original post, I think things have stabilized. I have not been able to reproduce any of these problems any longer. Also note that some of these problems, in particular if the above Bioconductor installation does not work for you, then you might also want to consult [the Bioconductor support site](https://support.bioconductor.org/), as it is likely a problem with your specific installation of R, etc. Note that the stable release is distributed as precomputed binaries for OS X and Windows, and only distributed after the package has passed all build checks via Bioconductor servers. 

I will close for now. Please post back if there are any lingering problems.

**Comment by Cecilia-Wang** [comment]
Hi there, I am having the same problem, not being able to use phyloseq without crashing R. System/software information:
OS X EL Capitan version 10.11.6 (problem still there when I update to macOS Sierra version10.12.1)
R version 3.3.2 (every single package i currently have is up to date)
I tried to reinstall the new version phyloseq_1.18.1 but failed. errors/warnings see the following
 ```
source('http://bioconductor.org/biocLite.R')
Bioconductor version 3.4 (BiocInstaller 1.24.0), ?biocLite for help
biocLite('phyloseq')
BioC_mirror: https://bioconductor.org
Using Bioconductor 3.4 (BiocInstaller 1.24.0), R 3.3.2 (2016-10-31).
Installing package(s) ‘phyloseq’

  There is a binary version available but the source version is later:
         binary source needs_compilation
phyloseq 1.18.0 1.18.1             FALSE

installing the source package ‘phyloseq’

trying URL 'https://bioconductor.org/packages/3.4/bioc/src/contrib/phyloseq_1.18.1.tar.gz'
Content type 'application/x-gzip' length 3291529 bytes (3.1 MB)
==================================================
downloaded 3.1 MB

* installing *source* package ‘phyloseq’ ...
** R
** data
** inst
** preparing package for lazy loading

 *** caught segfault ***
address 0x18, cause 'memory not mapped'

Traceback:
 1: dyn.load(file, DLLpath = DLLpath, ...)
 2: library.dynam(lib, package, package.lib)
 3: loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]])
 4: namespaceImport(ns, loadNamespace(i, c(lib.loc, .libPaths()),     versionCheck = vI[[i]]), from = package)
 5: loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]])
 6: namespaceImport(ns, loadNamespace(i, c(lib.loc, .libPaths()),     versionCheck = vI[[i]]), from = package)
 7: loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]])
 8: namespaceImport(ns, loadNamespace(i, c(lib.loc, .libPaths()),     versionCheck = vI[[i]]), from = package)
 9: loadNamespace(package, lib.loc, keep.source, partial = TRUE)
10: withCallingHandlers(expr, packageStartupMessage = function(c) invokeRestart("muffleMessage"))
11: suppressPackageStartupMessages(loadNamespace(package, lib.loc,     keep.source, partial = TRUE))
12: code2LazyLoadDB(package, lib.loc = lib.loc, keep.source = keep.source,     compress = compress)
13: makeLazyLoading(pkg_name, lib, keep.source = keep.source)
14: doTryCatch(return(expr), name, parentenv, handler)
15: tryCatchOne(expr, names, parentenv, handlers[[1L]])
16: tryCatchList(expr, classes, parentenv, handlers)
17: tryCatch(expr, error = function(e) {    call <- conditionCall(e)    if (!is.null(call)) {        if (identical(call[[1L]], quote(doTryCatch)))             call <- sys.call(-4L)        dcall <- deparse(call)[1L]        prefix <- paste("Error in", dcall, ": ")        LONG <- 75L        msg <- conditionMessage(e)        sm <- strsplit(msg, "\n")[[1L]]        w <- 14L + nchar(dcall, type = "w") + nchar(sm[1L], type = "w")        if (is.na(w))             w <- 14L + nchar(dcall, type = "b") + nchar(sm[1L],                 type = "b")        if (w > LONG)             prefix <- paste0(prefix, "\n  ")    }    else prefix <- "Error : "    msg <- paste0(prefix, conditionMessage(e), "\n")    .Internal(seterrmessage(msg[1L]))    if (!silent && identical(getOption("show.error.messages"),         TRUE)) {        cat(msg, file = stderr())        .Internal(printDeferredWarnings())    }    invisible(structure(msg, class = "try-error", condition = e))})
18: try({    suppressPackageStartupMessages(.getRequiredPackages(quietly = TRUE))    makeLazyLoading(pkg_name, lib, keep.source = keep.source)})
19: do_install_source(pkg_name, instdir, pkg, desc)
20: do_install(pkg)
21: tools:::.install_packages()
An irrecoverable exception occurred. R is aborting now ...
/Library/Frameworks/R.framework/Resources/bin/INSTALL: line 34:  1013 Done                    echo 'tools:::.install_packages()'
      1014 Segmentation fault: 11  | R_DEFAULT_PACKAGES= LC_COLLATE=C "${R_HOME}/bin/R" $myArgs --slave --args ${args}

The downloaded source packages are in
	‘/private/var/folders/kc/0ttbqsbj4vg65s_htf13qj1w9h3b9f/T/RtmpksQj5f/downloaded_packages’
Warning message:
In install.packages(pkgs = doing, lib = lib, ...) :
  installation of package ‘phyloseq’ had non-zero exit status`
```
Since that failed.
I tried to download phyloseq (phyloseq_1.18.1.tar.gz) from bioconductor and drag the unzipped folder in the default R Library path. After that I amw able to see it in R studio's available package.
However, whenever I try to use phyloseq (e.g library(phyloseq)), R studio will just quit itself. I tried the same thing in R console, which gives me the errors like the following.



```
     *** caught segfault ***
address 0x18, cause 'memory not mapped'

Traceback:
 1: dyn.load(file, DLLpath = DLLpath, ...)
 2: library.dynam(lib, package, package.lib)
 3: loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]])
 4: namespaceImport(ns, loadNamespace(i, c(lib.loc, .libPaths()),     versionCheck = vI[[i]]), from = package)
 5: loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]])
 6: namespaceImport(ns, loadNamespace(i, c(lib.loc, .libPaths()),     versionCheck = vI[[i]]), from = package)
 7: loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]])
 8: namespaceImport(ns, loadNamespace(i, c(lib.loc, .libPaths()),     versionCheck = vI[[i]]), from = package)
 9: loadNamespace(package, lib.loc)
10: doTryCatch(return(expr), name, parentenv, handler)
11: tryCatchOne(expr, names, parentenv, handlers[[1L]])
12: tryCatchList(expr, classes, parentenv, handlers)
13: tryCatch(expr, error = function(e) {    call <- conditionCall(e)    if (!is.null(call)) {        if (identical(call[[1L]], quote(doTryCatch)))             call <- sys.call(-4L)        dcall <- deparse(call)[1L]        prefix <- paste("Error in", dcall, ": ")        LONG <- 75L        msg <- conditionMessage(e)        sm <- strsplit(msg, "\n")[[1L]]        w <- 14L + nchar(dcall, type = "w") + nchar(sm[1L], type = "w")        if (is.na(w))             w <- 14L + nchar(dcall, type = "b") + nchar(sm[1L],                 type = "b")        if (w > LONG)             prefix <- paste0(prefix, "\n  ")    }    else prefix <- "Error : "    msg <- paste0(prefix, conditionMessage(e), "\n")    .Internal(seterrmessage(msg[1L]))    if (!silent && identical(getOption("show.error.messages"),         TRUE)) {        cat(msg, file = stderr())        .Internal(printDeferredWarnings())    }    invisible(structure(msg, class = "try-error", condition = e))})
14: try({    attr(package, "LibPath") <- which.lib.loc    ns <- loadNamespace(package, lib.loc)    env <- attachNamespace(ns, pos = pos, deps)})
15: library(phyloseq)
```

Any idea of how I can fix it?

**Comment by Cecilia-Wang** [comment]
I had the same problem with R 3.3.2 MacOS 10.11 (El Capitan) and MacOS 10.12.1 (Sierra). I got  warnings/errors with 15 to 20 Traceback starts like this:
```
     *** caught segfault ***
address 0x18, cause 'memory not mapped'
```
I did lots of things trying to fix it, such as updating all the packages and reinstall R, R studio, Xcode and XQuartz, but all failed.
The final solution I got was to **delete R, all the R history files and all the library files. Note that uninstall R does not automatically remove the R history or the library files for you.**
Then reinstall everything, and yes you will need to reinstall every single package that is not default in R.
Hope this will help someone :)
